### PR TITLE
Remove unlinked a user if the user is deleted or removed from a site.

### LINKED ISSUE #2188: Deleted user but not disconncted from .com
Currently if a user is connected to .com but gets deleted before disconnecting first, the user is in a limbo state where they can't disconnect Jetpack from Calypso but can't do anything with that site since they don't have permissions, since they are not a valid user anymore. 

I think we need to fix this in Jetpack to help avoid this situation by disconnecting users when they are deleted.

### PR DESCRIPTION
**Pull request by enejb**
Partially fixes #2188. We still need to take the the master user into account here. 

I think the fix would need to be different.

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -470,6 +470,10 @@ private function Jetpack() {
 		add_action( 'user_register', array( $this, 'is_single_user_site_invalidate' ) );
 		add_action( 'deleted_user', array( $this, 'is_single_user_site_invalidate' ) );
 
+		// Unlink user before deleting the user from .com
+		add_action( 'deleted_user', array( $this, 'unlink_user' ), 10, 1 );
```
Comment (dereksmart) [inline class.jetpack.php]: `user_role_change()` might help here - especially if the user you're deleting is the master user.

Comment (enejb) [inline class.jetpack.php]: How would the user_role_change help?. Also unlink_user doesn't work for the master user. That case would have to be fixed separately.

Comment (dereksmart) [inline class.jetpack.php]: I mean hook `user_role_change` onto `deleted_user` - I think it will re-assign the master user if the deleted user is master.

Comment (enejb) [inline class.jetpack.php]: I think we need to fix the master user connection in a better way. With some sort of UI element.

**Comment by enejb** [comment]
I think this should also be part of 3.6. cc @georgestephanis and @dereksmart :)

**Comment by dereksmart** [comment]
Merging, and we can take care of the master user deletion in another way (likely a warning in the new My Jetapck page)

**Pull request merged**